Assets/Scripts/Combat/Gun.cs

```diff
@@ -30,14 +30,16 @@ void Update()
             {
                 timer = 0f;
                 FireGun();
-                muzzleFlash.Play();
             }
         }
     }
 
+    //casts a ray that will collide with objects and detract health if they have any
     private void FireGun()
     {
-        Debug.DrawRay(firePoint.position, firePoint.forward * 100, Color.blue, 2f);
+        Debug.DrawRay(firePoint.position, firePoint.forward * 100, Color.red, 2f);
+
+        muzzleParticle.Play();
 
         Ray ray = new Ray(firePoint.position, firePoint.forward);
```

Comment:
You can add a parameter for your ray for ray distance - it might be a good idea to create a serialized firingRange variable and add this as a parameter